### ISSUE TITLE
Sri Lanka Holiday Calendar Feed

### ISSUE THREAD

**Issue by pathumego** [CodeLanka/codeLanka-Proposals#14]
This is a really small project, but will be really really useful.. a Public Google Calendar could be used I guess. 

Or can this be on GitHub pages and serve the iCal  (https://github.com/budparr/jekyll-calendar)

**Comment by pasindud** [comment]
Yea this should be interesting mini project

**Comment by pathumego** [comment]
i will do the input if someone can set up the site :)

**Comment by pasindud** [comment]
I'll setup the site 
https://github.com/CodeLanka/sri-lanka-calendar

**Comment by dhammika-marasinghe** [comment]
I am maintaining a public calender of Sri Lanka holidays for my use. Hope we can use it. :)
https://www.google.com/calendar/embed?src=kr9fgv9anqi01er9aua1a8bd3s%40group.calendar.google.com&ctz=Asia/Colombo

**Comment by pasindud** [comment]
@dhammika-marasinghe  can you make a small script that fetch data from that xml and writes to https://github.com/budparr/jekyll-calendar/blob/master/_data/events.yaml

**Comment by ishan-marikar** [comment]
This might be of interest to you: https://db.tt/5S4dVa54 .. I use it with my outlook.

**Comment by ishan-marikar** [comment]
I'll do a bit of research on the structure of ical files and then try my hand at coding a parser for it.

**Comment by ishan-marikar** [comment]
Well, instead of parsing it, I just converted it into a csv file and then converted that into JSON/XML.

**Comment by pasindud** [comment]
That works too, the JSON schema must be bit agile, can guys post a example JSON with one event and the metadata.

**Comment by ishan-marikar** [comment]
```javascript
[
        {
            title  : 'National Day',
            description : "Independence day of Sri Lanka",
            date: '2015-02-04'
        }
] 
```

what about that?

**Comment by pasindud** [comment]
Start date end Date ( for incase )
Type - Mercantile || Bank  (important)

I think this is ok for a base

**Comment by pathumego** [comment]
Hi Guys!

I think rather than maintaining a Google calendar its better to have this
on GH with GH pages serving the website and the iCal feed, if possible.
That way anyone can fork the repo add a calendar days and send a PR. Or can
use issues.

And options to modify the feed to include the needed days would be awesome.

- Holidays Public
- Holidays Bank
- Holidays Merc.

Other educational and recreational calendars

- Important National Days / Weeks
- Important birthdays
- History

These can be important for apps and blogs.

**Comment by pasindud** [comment]
The only reason for Google Calendar is to allow non tech people to contribute too, and google kind dose the work of generating xml, json and Ical

**Comment by pasindud** [comment]
JSON google calendar https://www.google.com/calendar/feeds/kr9fgv9anqi01er9aua1a8bd3s%40group.calendar.google.com/public/basic?&alt=json

**Comment by pasindud** [comment]
https://github.com/CodeLanka/common-data-sri-lanka/pull/6

**Comment by pathumego** [comment]
ok I agree, how about rest of the suggestions? 

```
Options to modify the feed to include the needed days would be awesome.

- Holidays Public
- Holidays Bank
- Holidays Merc.

Other educational and recreational calendars

- Important National Days / Weeks
- Important birthdays
- History

```

**Comment by ishan-marikar** [comment]
@pathumego @pasindud What about an API that basically does that?
Something like:

../calendar?calendar_type=history&holiday_type=public&from=01011992&to=31122015

(Show all holidays from the History calendar where the holidays are public, within the range of 01st of January, 1992 until 31st December, 2015.

**Comment by pasindud** [comment]
Lets keep holidays file and history as two (separation of concerns)

```
/history 

from
to
```

```
/holiday

from
to
holiday_type
```


API should also have paging to deal with large set of data.

**Comment by ishan-marikar** [comment]
calendar/holidays/?from=01011992&to=31122015
calendar/history/?from=01011992&to=31122015

**Comment by pathumego** [comment]
@pasindud  @ishan-marikar guys Im not very well informed about technology, but I can comment on the users angle or UX of the project :)

**Comment by pathumego** [comment]
ok so to be clear I think there should be 3 kinds (or even more custom kinds?) of events on the non-holiday section as mentioned above.

**Comment by pasindud** [comment]
you mean something called "other calendar", where stuff that is not a holiday or history goes to ?. if so Good Idea works well :+1:

**Comment by pathumego** [comment]
well, not really, The definition/contents of 'history' is not clear here i guess. 

What is clear (and a separate calendar for sure) is - Important National Days / Weeks calendar.( ie: National Dengue Day week, Scouts Job Week)

Rest could be other calendar/ calendars as @pasindud suggests

**Comment by pasindud** [comment]
Let's maybe work on the holiday and history first to see how this works out.

**Comment by pasindud** [comment]
@dhammika-marasinghe @ishan-marikar  can you guys look into the tech side of this think u have a commit access to the repo's, if not let me know

**Comment by prasadsw** [comment]
@pasindud What's the status of this project? do we have a proper iCal calendar in place? are we regularly updating it?

**Comment by pathumego** [comment]
@pahans @pasindud  What do we do about this?

**Comment by pasindud** [comment]
I think @ishan-marikar was going to do something about this

**Comment by ishan-marikar** [comment]
Yes, I was going to do something about this, but I don't really know what I should do. .___.
